Chemical Prioritization System - ChemPSy
==========
[//]: # (https://rozaxe.github.io/factory/)

[![Stable v1.0.0](https://img.shields.io/badge/Stable-v0.1.0-green.svg)]()
[![Python v>=2.7.12](https://img.shields.io/badge/Python-v>=2.7.12-0077ea.svg)](https://www.python.org/)
[![License: GPL-3](https://img.shields.io/aur/license/yaourt.svg)](https://opensource.org/licenses/GPL-3.0)The aim of **ChemPSy** (Chemical Prioritization System) is to develop an innovative approach based on several bioinformatics and biostatistics methodologies to analyze and integrate massive toxicogenomics datasets. Specific objectives include: (1) classification of chemicals based on transcriptional signatures, e.g. the set of genes whose expression is known to be positively or negatively altered after an exposure to these compounds; (2) the association of classes with human pathologies or deleterious phenotypes, e.g. classes containing toxicants with well-known effects; (3) the prediction of novel reprotoxicants and/or endocrine disruptors based on transcriptional signature similarities with known chemicals affecting testis development and function.##Data formatEach dataset is organizing according to the following:   	+-- [Species]         +-- [Tissue]    		+--GSEXXX    			+--Experimental_conditions    			¦	+--Condition 1    			¦	+--Condition 2    			+--Individual_experiments    			¦	+--GSMXXXX.CEL.gz    			¦	+--GSMXXXX.CEL.gz      			+--GSEXXX.txt  **[Species]**: Binomial nomenclature for selected species (e.g. Homo sapiens for Human)**[Tissue]**: Tested tissue in upper case (e.g. LIVER, KIDNEY or MCF-7, HK-2...)___###Step 1 - Describe your datasetTo describe your dataset please use tabulate .txt file with the following fields (**Keep the order**):| Fields | Description || ------ | ----------- || Files   | CEL file full name (GSM1223.CEL.gz) || Species | Binomial nomenclature of species where the results come from || Strain    | Species strain (e.g. Sprague-Dawley). _Can be not specified_ * || Gender    | Animal gender(male/female). _Can be not specified_ * || Experiment    | Experiment (E.G. _in vitro_, _in vivo_, ...). _Can be not specified_ * || Tissues/Cells    | Tissue or cell name where the experiment is performed || Age    | Animal age. _Can be not specified_ * || Generation    | Animal generation (for trans-generational studies). If not specified, please put 'F0'.  || ChemicalName    | Chemical usual / synonym name (only one name) || CAS	   | Chemical CAS number |``| MESH    | Chemical MESH ID || Dose    | Chemical exposition dose || Duration    | Chemical exposition duration || Route    | Chemical route. _Can be not specified_ * || Vehicule    | Chemical vehicle. _Can be not specified_ * || PMID    | Associated publication PubMed ID. _Can be not specified_ * || GSE    | GEO dataset ID || GSM    | GEO profile ID || GPL    | GPL use. _Can be not specified_ * || Mail    | Corresponding dataset author mail. _Can be not specified_ *  || Paired    | Paired data (Yes/No) || Replicates    | Replicate number || Experiment type    | Experimental type details (e.g. 'Expression profiling by array’). _Can be not specified_ * || Design    | Experimental design. _Can be not specified_ * || Treatment protocol   | Treatment protocol description. _Can be not specified_ * || Characteristics    | Tissue/cells characteristics. _Can be not specified_ * || Extraction protocol   | Extraction protocol description. _Can be not specified_ * || Link(s) | Cross-link(s) (e.g. GEO, database, personal website ...). _Can be not specified_ * || Data processing    | Data processing description. _Can be not specified_ *|| Sample	Treated   | Treated or Control sample. _Can be not specified_  || Associated Ctrl  | Associate a unique number to your control and list all control paired with your treated sample (e.g. control1 = 1, control2=2 ..., treated_sample1 = 1,2 [this sample is paired with control 1 and 2]). |**Don't leave empty fields**: use 'NA' if your field is not specified  **'*'**: This field is required for TOXsIgN integration  Each line need to correspond to one and unique sample###Step 2 - Organize your dataIn your GSEXXX directory, save your tabulate .txt file using the same name of your directory: GSEXXX.txt and create a new folder called: _Individual\_experiments_.  Drop in this folder all expression files associated with your study. Please make sur that all yours. CEL file are compressed. If not use the following command:      gzip *.CEL    ###Step 3 - Create conditions and treatment.infoTo create the Experimental_conditions directory and all conditions sub-directories, use the _**CreateTreatmentInfo.sh**_ script. This script takes no arguments but load a configuration file: _**ChemPSy.ini**_. Please modified this file or change the configuration file load in _**CreateTreatmentInfo.sh**_ script:	source /home/genouest/irset/tdarde/projects/ChemPSy/20160321/script/ChemPSy_Human.iniNext adapt the loop according to your datasets:	#! /bin/bash	source /home/genouest/irset/tdarde/projects/ChemPSy/20160321/script/ChemPSy_Human.ini	echo "Reading config...." >&2	#echo "Create treatment.info files for HEPATOCYTES"	for i in $HepatoList	    do	        python $scriptTreatment -p $i -t HEPATOCYTES -e $fileRemove -s True	done		echo "Create treatment.info files for HK-2"	for i in $HK2List	    do	        python $scriptTreatment -p $i -t HK-2 -e $fileRemove -s True	done		echo "Create treatment.info files for ISHIKAWA_CELLS"	for i in $IshikawaList	    do	        python $scriptTreatment -p $i -t ISHIKAWA_CELLS -e $fileRemove -s True	done		echo "Create treatment.info files for JURKAT_CELLS"	for i in $JurkatList	    do	        python $scriptTreatment -p $i -t JURKAT_CELLS -e $fileRemove -s True	done		echo "Create treatment.info files for MCF-7"	for i in $MCFList	    do	        python $scriptTreatment -p $i -t MCF-7 -e $fileRemove -s True	done		echo "Create treatment.info files for liver"	for i in $LiverList	    do	        python $scriptTreatment -p $i -t LIVER -e $fileRemove -s False	done		echo "Create treatment.info files for tg"	for i in $TgList	    do	        python $scriptTreatment -p $i -t THIGH-MUSCLE -e $fileRemove	doneIf you have no error, you may obtain the following directories organization:	+-- [Species]         +-- [Tissue]    		+--GSEXXX    			+--Experimental_conditions    			¦	+--Condition 1    			¦	¦	+--treatment.info    			¦	+--Condition 2    			¦	¦	+--treatment.info    			+--Individual_experiments    			¦	+--GSMXXXX.CEL.gz    			¦	+--GSMXXXX.CEL.gz      			+--GSEXXX.txt  In each _treatment.info_ you will find the association between treated sample (first column) and control sample (second column):	003016029014.CEL.gz	003016029008.CEL.gz	0	003016029014.CEL.gz	003016029009.CEL.gz	0	003016029015.CEL.gz	003016029008.CEL.gz	0	003016029015.CEL.gz	003016029009.CEL.gz	0																		##Run ChemPSy**Before run ChemPSy please be sur that you have the same architecture like previously describe and you have all your conditions with associated treatment.info files**Next run **_ChemPSy_data_prep.sh_**  As the previous script, this script uses the same configuration file. So please edit it and/or change the path on source line.	#!/bin/bash	#################################	#     Source .ini file          #	#################################		echo "##############################       ChemPSy       ##############################"			echo "--1-- Checking config file"	source /home/genouest/irset/tdarde/projects/ChemPSy/20160321/script/ChemPsy_processing/ChemPSy_Human.ini	echo "Reading config...." >&2	echo "Reading scriptPath: $scriptPath" >&2	echo "Reading config: $Rscript " >&2	echo "Reading dataPath: $dataPath" >&2	echo "Reading config: $processedPath " >&2###Step 1 - Quality controlThe first step of **_ChemPSy_data_prep.sh_** is a quality control. Various information will be created for each conditions including the microarray picture. 	function step_1 {		echo "--2-- STEP_1 process_data"		for tissue in $tissues			do				echo $tissue				outputT=$processedPath$tissue"/"				mkdir -p $outputT								for gse in $gsePath					do						path=$dataPath$tissue"/"$gse"/"						if [ -d $path ]							then								output=$processedPath$tissue"/"$gse"/Experimental_conditions/"								mkdir -p $output								scriptA=$scriptPath'Rlauncher.sh'								$scriptA -p $path -t $tissue -o $output -c $cdfpath						fi				done		done		while [ $(qstat | grep "ChemPSy_" | wc -l) -ne 0 ]		do			echo "Running --2-- STEP_1 process_data"			sleep 7		done		echo "--2-- STEP_1 process_data finish"	} To performe the quality control please check each picture one by one and remove microarray with 20% or more of hybridization error.  List all your sample to remove in **_removeCelFile.txt_** and run the **Step - 3** of Data Format part  	+-- [Species]         +-- [Tissue]    		+--GSEXXX    			+--Experimental conditions    				+--Condition 1    				¦	+--contrastmatrix.txt     				¦	+--normdata.txt     				¦	+--designmatrix.txt    				¦	+--qc_boxplot_afternormalization.pdf    				¦	+--qc_boxplot_beforenormalization.pdf    				¦	+--filtration.txt     				¦	+--log2fcchangedata.txt    				¦	+--qc_corrmatrix_afternormalization.pdf    				¦	+--mednormdata.txt    				¦	+--qc_image_003016029009.CEL.gz.png    				¦	+--qc_image_003016029015.CEL.gz.png    				¦	+--qc_image_003016029009.CEL.gz.png###Step 2 - List all conditionsThis step lists all the conditions 